### DISCUSSION TITLE
SonarScanner execution

### DISCUSSION THREAD

**Discussion by Sharan1808** [learndevopsonline/learndevopsonline#1810] [Q&A]
ERROR: Error during SonarScanner execution
java.lang.UnsupportedClassVersionError: org/sonar/batch/bootstrapper/Environmentrsion of the Java Runtime only recognizes class file versions up to 55.0

I'm getting the above error in D76-2023-12-22-SESSION-53,  my java version is Java 11.0.3, I tried different ways but got the same error.

sonarqube public IP : 44.220.171.156
workstation IP: 54.197.10.118 

![Screenshot (398)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/94f7b251-1167-4825-9813-0e3af5ee13a5)

**Comment by chaithanya1812** [comment]
Hi Sharan,
I tried with sonar-scanner-5.0.0 version, it's working but it's need disk space.
can you increase disk size check once?
![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/09923c82-54b6-42e3-8a12-7b9b4393c7d0)

**Reply by Sharan1808** [reply]
Thanks for the info I have added disk space as you suggested 

But I'm getting some other errors.

Workstation IP:  34.238.161.190
Sonarqube IP:   35.170.68.175
Jenkins IP:         50.16.10.3

![Screenshot (399)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/a621e87d-27d4-4fd3-94c7-4f2c9e8b1583)
![Screenshot (400)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/9928865f-af71-40ef-af34-fe9a78201f67)
![Screenshot (401)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/b7a1a047-1f49-4c0e-808a-83a431333bb0)
![Screenshot (402)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/a1c63823-3e9e-46a2-a693-76acab91ab23)
![Screenshot (403)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/4a16b7f1-acc8-42a1-95a4-6d024bb80027)
![Screenshot (404)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/6d7660a1-778f-4962-804f-0c084d212c87)

**Comment by chaithanya1812** [comment]
Hi Sharan,

I exclude JS/TS files from your analysis using the 'sonar.exclusions' configuration property. so it's working fine.
 
But ask once this to raghu sir is this correct or not?

/tmp/sonar-scanner-5.0.0.2966-linux/bin/sonar-scanner -Dsonar.host.url=http://172.31.10.101:9000 -Dsonar.token=squ_2b5822337c5a683ab4164328e0c5fb78696e09d5 -Dsonar.projectKey=expense-backend -Dsonar.projectName=expense-backend  -Dsonar.javascript.exclusions="*.js"

![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/508b77a1-98ec-44ba-b09a-e1a698d7faa1)

**Reply by Sharan1808** [reply]
Hi Chaitanya,

I'm getting the same error again can you please confirm with me one more time.

![Screenshot (405)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/5e7206f0-ff5e-4175-9e43-cd46188b4b93)

**Comment by chaithanya1812** [comment]
Try once this command
/tmp/sonar-scanner-5.0.0.2966-linux/bin/sonar-scanner -Dsonar.host.url=http://172.31.10.101:9000/ -Dsonar.token=squ_2b5822337c5a683ab4164328e0c5fb78696e09d5 -Dsonar.projectKey=expense-backend -Dsonar.projectName=expense-backend -Dsonar.javascript.exclusions="*.js"

**Reply by Sharan1808** [reply]
Thank you so much it's finally working now.

But when we check in SonarQube it's showing as no lines of code.

![Screenshot (406)](https://github.com/learndevopsonline/learndevopsonline/assets/148907638/1f9683e8-d25d-4947-935d-5c60e4d87fb8)

**Reply by chaithanya1812** [reply]
can you join  lab session today.

**Reply by Sharan1808** [reply]
ok sure
